backend/src/nodes/impl/stable_diffusion/stable_diffusion.py

```diff
@@ -0,0 +1,33 @@
+from typing import List
+import numpy as np
+import sdkit
+import torch
+from PIL import Image
+from sdkit.models import load_model, unload_model
+from sdkit.generate import generate_images
+
+
+class StableDiffusion:
+    def __init__(self, context: sdkit.Context):
+        self.context = context
+
+    @classmethod
+    def from_file(cls, path: str):
+        context = sdkit.Context()
+        context.model_paths["stable-diffusion"] = path
+        return cls(context)
+
+    @torch.inference_mode()
+    def forward(self, prompt: str) -> np.ndarray:
+        load_model(self.context, "stable-diffusion")
+
+        images: List[Image] = generate_images(context=self.context, prompt=prompt)
+        image = np.array(images[0])
+
+        unload_model(self.context, "stable-diffusion")
+
+        # XXX TODO Would be very nice to not load and unload the model completely every time. Need to figure out a
+        #  way to load_model an sdkit.Context to cpu, pass that around, and send it to the gpu when we want to run.
+        #  like ONNX or PyTorch
```

Comment:
Maybe what might be a good thing to attempt with this is actually just putting all the necessary code into chainner itself. We tend to do this with anything we can do we can 
1) cut down on the number of dependencies needed 
2) modify things as necessary 

For example, this library installs a bunch of unnecessary stuff like the realesrgan and gfpgan packages. We could just remove everything pertaining to that since we don't need those. 

And of course part of that could be implementing what this comment says (if it's not possible with what's available)

Reply:
The other thing that would be nice is being able to parse the state dict to determine what version of the model it is, if the arch changed between versions. I do this with all the other models. Only thing is, i have no idea if that's possible with .ckpt files and it's part of why i gave up initially on implementing SD :/

Reply:
Would it make sense to fork `sdkit` and make the changes we need to that?  If we just copy code around, it'll be a much more manual process to include upstream improvements.

Reply:
Yeah, in this case that would make more sense. Good idea